File: 1749-maximum-absolute-sum-of-any-subarray/1749-maximum-absolute-sum-of-any-subarray.java
class Solution {
    public int maxAbsoluteSum(int[] nums) {
        int sum = 0, maxSum = 0, minSum = 0;
        for(int num : nums) {
            sum += num;
            maxSum = Math.max(maxSum, sum);
            minSum = Math.min(minSum, sum);
        }
        return maxSum - minSum;
    }
}

File: 1749-maximum-absolute-sum-of-any-subarray/1749-maximum-absolute-sum-of-any-subarray.py
class Solution:
    def maxAbsoluteSum(self, nums: List[int]) -> int:
        sums, maxSum, minSum = 0, 0, 0
        for num in nums:
            sums += num
            maxSum = max(maxSum, sums)
            minSum = min(minSum, sums)
        return maxSum - minSum